lib/auth/sessions.go

```diff
@@ -103,6 +109,49 @@ func (s *Server) CreateAppSession(ctx context.Context, req types.CreateAppSessio
 	if err = s.UpsertAppSession(ctx, session); err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	// User metadata from the audit event
+	userMetadata := identity.GetUserMetadata()
+	userMetadata.User = session.GetUser()
+	userMetadata.AWSRoleARN = req.AWSRoleARN
+
+	// Record peer for the address of the requesting connection.
+	p, ok := peer.FromContext(ctx)
+	if !ok {
+		return nil, trace.Wrap(errors.New("unable to get peer from context"))
+	}
+
+	// Now that the certificate has been issued, emit a "new session created"
+	// for all events associated with this certificate.
+	appSessionStartEvent := &apievents.AppSessionStart{
```

Comment:
It looks like something might be slightly off. I rebuilt tsh and teleport, restarted my cluster and then logged in again, then proceed to login to my app through tsh. In the audit log I saw "User [rav] has connected to application [undefined]".

I'm not sure if something is wrong with the code or my setup but logging in through the Web UI produces an event with a correct app name.

Reply:
Thank you, good catch. I'll check this out.

Reply:
Ah, sorry, I forgot that I installed the tsh.app from our downloads page a couple of weeks ago and it overshadowed my usual setup that was to use `tsh` from the build folder.

I just tested it with tsh from your branch and everything works as expected. 👍

Reply:
Awesome, thank you! :-) I appreciate it!